### ISSUE TITLE
Enabling "Send notification on button press or IR" does not respect UI Sync status

### ISSUE THREAD

**Issue by blazoncek** [Aircoookie/WLED#3360]
### What happened?

When you enable "Send notification on button press or IR" in Sync settings UDP message is sent regardless of Sync being activated or not.

This behaviour adds ambiguity to how Sync works as a user may expect that turning Sync on (in main UI) will enable sync notifications and deactivating it will disable all sync notifications. While settings will define which of the events are causing sync notifications.

### To Reproduce Bug

Enable "Send notification on button press or IR" in Sync settings but do not activate Sync in main UI.

### Expected Behavior

Discussion with some users ended with 2 possible outcomes:
a) remove main UI Sync button and rely on settings alone
b) use main UI Sync button as a master toggle if notifications are sent or not, for activators rely on settings

There may also be no need to differentiate between activators, i.e. direct change (UI), button/IR, Hue, etc.

I am leaning toward option b) as IMO the ability to toggle sync on and off from UI may be beneficial.

### Install Method

Self-Compiled

### What version of WLED?

0.11+ (using 0.14.0-b4)

### Which microcontroller/board are you seeing the problem on?

ESP8266, ESP32, ESP32-S2, ESP32-C3

### Relevant log/trace output

```shell
N/A
```

### Anything else?

https://discord.com/channels/473448917040758787/1149007835166937210

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Aircoookie** [comment]
This is intended behavior as the main UI button is only supposed to toggle sync on direct changes (use case: you toggle a group of devices with a button but still want to set the device individually via UI/API without having to disable and re-enable sync each time). 

Though I can see how this might cause confusion, so I'd be open for the behavior to be changed to proposal B.

**Comment by blazoncek** [comment]
> This is intended behavior

Thanks @Aircoookie I am well aware of that. 😁 The issue is that it is unintuitive as far as settings go.
When you put those checkmarks in settings, you may not expect WLED to start syncing immediately. Normal (or casual) user might expect that syncing only start when you select main UI button which (very obviously) shows that syncing is active.

**Comment by blazoncek** [comment]
There is also another issue with how sync works:
If I have a master device (*to rule them all*) equipped with IR (or ESP-NOW) and MQTT (etc) and I want it to sync with slaves, I would enable button/IR sync in setting. This would propagate changes made using buttons/IR/ESP-NOW but not UI or MQTT while both change WLED state in the same way and UI will not show sync being active. Perhaps I fail to see the difference why would you want button/IR to always sync (if enabled in settings) while UI changes only after clicking main Sync button (UI element). This is also inconsistent if you just observe UI itself (it will react to any change).

I know this is re-iteration from above but it is seen from different perspective.

**Comment by Aircoookie** [comment]
The idea is that while I think it's common to use sync to switch all WLED lights with IR/espnow/huesync, if you go through the trouble of going to the "master" lights UI, you may want to toggle it individually. 

I agree that having the UI button as a master sync button would be intuitive, so it's ok to change this. There are other ways to achieve that usecase, for example turning off direct change sync in settings on the button device and using a different device as "master" if wishing to set the entire group via UI.

If current behavior was retained, we should at least change the tooltip text when toggling the syc button to "Changes to this light via UI will (now/no longer) sync with other lights" to make it clearer how it works, since the current tooltip indeed suggests it would toggle sync entirely.

**Comment by blazoncek** [comment]
I've implemented global sync toggle as described in b) above and behaviour seems more intuitive (at least to me and my family).

Fiddling still further with sync, there is a feature of receive sync hidden in UI settings. I am not sure I understand its function correctly: When unchecked, WLED will *always* receive UDP sync regardless of UI sync button. But when checked, UI sync button will toggle sending and receiving at the same time. I find such behaviour even more confusing, if sync is off, WLED will neither send nor receive UDP notifications and if it is on, it will broadcast and receive at the same time. What is the purpose of such behaviour?

**Comment by Aircoookie** [comment]
Cool, behavior described in b) seems good!

You understood correctly, the option toggles between the main UI sync button toggling sending only (leaving receiving up completely to what is set in settings) and toggling both send + receive.

I do not personally use this option, IIRC someone requested it a long time ago. The default behavior of the sync button to toggle sending is the most intuitive in my opinion. It would be a pain to have to remember to toggle off sync and toggle it back on every time you change a device but still want it to receive sync.

Fun fact: WLED up to 0.8 (old UI) had 2 sync buttons to toggle send and receive separately. But for receive at least I think it's fine to stick to having it in settings.

**Comment by blazoncek** [comment]
Since we now have sync groups it may be irrelevant to have separate "receive sync" button as you can effectively remove all groups and WLED will not receive anything.
The same would also be true for sending. If you uncheck all sending groups, WLED will not send out anything.
